### ISSUE TITLE
Cannot reproduce RoboTwin results when training from scratch — only difference vs. official is global batch size

### ISSUE THREAD

**Issue by PKUduz** [yuantianyuan01/FastWAM#65]
## Summary
I trained from scratch using the official config `robotwin_uncond_3cam_384_1e-4` (softmax / 3-cam / 384). Training loss converges normally, but my RoboTwin success rates are well below the paper (reported ~91.8% overall; see #23).

The **only difference from the official setup is the global batch size**: I'm limited to 2 nodes x 8 GPUs, per-GPU batch = 16 -> **global batch = 256**, whereas the official setup is 8 nodes x 8 GPUs -> **global batch = 1024**. Everything else (LR = 1e-4, `gradient_accumulation_steps=1`, 3-cam, 384, dataset) is identical. I've also matched the dependency versions from the #23 fix (`sapien==3.0.0b1`), so this is not the known sapien rendering issue.

## Reproduction results (self-trained checkpoint, step 117415, 100 episodes/task)

| Task | Clean | Random |
|---|---|---|
| adjust_bottle | 0.98 | 0.97 |
| beat_block_hammer | 0.78 | 0.69 |
| blocks_ranking_rgb | 0.43 | 0.42 |
| blocks_ranking_size | 0.37 | 0.36 |
| click_alarmclock | 0.93 | 0.82 |
| click_bell | 0.99 | 0.91 |
| dump_bin_bigbin | 0.93 | 0.90 |
| grab_roller | 0.99 | 0.99 |
| handover_block | 0.74 | 0.57 |
| handover_mic | 0.94 | 0.88 |
| lift_pot | 0.39 | 0.37 |
| move_can_pot | 0.43 | 0.51 |
| move_playingcard_away | 0.84 | 0.90 |
| move_stapler_pad | 0.25 | 0.15 |
| hanging_mug | 0.04 | 0.04 |
| open_laptop | 0.65 | 0.66 |
| open_microwave | 0.14 | 0.18 |
| pick_diverse_bottles | 0.72 | 0.61 |
| pick_dual_bottles | 0.91 | 0.83 |
| place_a2b_left | 0.63 | 0.65 |
| place_a2b_right | 0.70 | 0.68 |
| place_bread_basket | 0.70 | 0.66 |
| place_bread_skillet | 0.67 | 0.61 |
| place_can_basket | 0.59 | 0.64 |
| place_cans_plasticbox | 0.80 | 0.89 |
| place_container_plate | 0.93 | 0.99 |
| place_dual_shoes | 0.58 | 0.59 |
| place_empty_cup | 0.93 | 0.94 |
| place_fan | 0.63 | 0.49 |
| place_burger_fries | 0.88 | 0.91 |
| place_mouse_pad | 0.29 | 0.27 |
| place_object_basket | 0.59 | 0.68 |
| place_object_scale | 0.61 | 0.63 |
| place_object_stand | 0.79 | 0.81 |
| place_phone_stand | 0.54 | 0.51 |
| move_pillbottle_pad | 0.69 | 0.70 |
| place_shoe | 0.84 | 0.80 |
| press_stapler | 0.54 | 0.61 |
| put_bottles_dustbin | 0.72 | 0.80 |
| put_object_cabinet | 0.62 | 0.65 |
| rotate_qrcode | 0.46 | 0.56 |
| scan_object | 0.55 | 0.47 |
| shake_bottle | 1.00 | 0.97 |
| shake_bottle_horizontally | 1.00 | 1.00 |
| stack_blocks_three | 0.51 | 0.58 |
| stack_blocks_two | 0.89 | 0.93 |
| stack_bowls_three | 0.61 | 0.66 |
| stack_bowls_two | 0.83 | 0.80 |
| stamp_seal | 0.55 | 0.47 |
| turn_switch | 0.40 | 0.42 |
| **Overall (50 tasks)** | **67.04%** | **66.26%** |

## Training config
- per-GPU `batch_size=16`, `num_workers=8`, `gradient_accumulation_steps=1`, `learning_rate=1e-4`
- **global batch = 256** (2 nodes x 8 GPU); official = **1024** (8 nodes x 8 GPU)

## Environment
- GPUs: 8x NVIDIA L20X (143 GB), Driver 570.148.08, CUDA 12.8 (training on 2 nodes x 8 GPUs)
- Python 3.11.15
- `torch==2.7.1+cu128`, `torchvision==0.22.1+cu128`, `triton==3.3.1`
- `transformers==4.49.0`, `diffusers @ git+https://github.com/huggingface/diffusers@f7fd76a`, `accelerate==1.12.0`
- `sapien==3.0.0b1`, `mplib==0.2.1`, `toppra==0.6.3`, `gymnasium==0.29.1`, `numpy==1.26.4`

## Questions
1. Is this ~25% gap mainly attributable to the smaller global batch (256 vs 1024), or is there another key setting I'm missing?
2. **Could you release the RoboTwin training curves** (loss, and success rate at intermediate checkpoints/steps)? That would let me tell whether this is under-convergence, small-batch gradient noise, or a config mismatch -- and avoid unnecessary re-training.

**Comment by zijianzhang30** [comment]
Encountered the same issue.

**Comment by AIprogrammer** [comment]
Hi, would you be able to share the training curves? It would be very helpful for understanding the training behavior.

**Comment by PKUduz** [comment]
> Hi, would you be able to share the training curves? It would be very helpful for understanding the training behavior.

<img width="2840" height="696" alt="Image" src="https://github.com/user-attachments/assets/296fdb7e-d44b-4a7c-ab39-4213a9b0edbf" />

**Comment by XindaQ** [comment]
Hi, how long have you used for the training (100k steps)? I would like to compare it with my case (also used 2 nodes).

**Comment by PKUduz** [comment]
> Hi, how long have you used for the training (100k steps)? I would like to compare it with my case (also used 2 nodes).

3 days

**Comment by in-situflash** [comment]
Have you solved the problem?

**Comment by PKUduz** [comment]
> Have you solved the problem?

NO

**Comment by in-situflash** [comment]
I tried to train FastWAM with a global batch size of 40 due to the limited GPUs. When training 40000 steps, my results were already better than yours. I only tested the task 'blocks_ranking_size', which is a relatively difficult task in RoboTwin 2.0, and got a success rate  of 65%. The training is still ongoing.

**Comment by LoveJu1y** [comment]
@in-situflash hello，have you reproduced the official result？

**Comment by LoveJu1y** [comment]
@PKUduz have you reporduced the official result now?

**Comment by yuantianyuan01** [comment]
Thanks for the detailed information. In your setting, the total number of training samples is roughly aligned with ours: 117k steps with global batch size 256 is close to 29.5k steps with global batch size 1024.

So I am not sure about the exact cause of this gap. It may be related to the change in global batch size and the corresponding adjustment of training steps, since this changes the optimization dynamics even when the total number of samples is similar.

We can provide the training curves for reference, and I hope they will be helpful. If possible, you may also try using gradient accumulation to make the effective global batch size equivalent to 1024 and see whether this improves the results.

<img width="1433" height="614" alt="Image" src="https://github.com/user-attachments/assets/61c8ed62-91ba-49a2-8131-bd3c726636ab" />
